src/simulation/LoadGenerator.cpp

```diff
@@ -458,43 +498,16 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
         }
     }
 
-    if (!mLoadTimer)
-    {
-        mLoadTimer = std::make_unique<VirtualTimer>(mApp.getClock());
-    }
-
     mStarted = true;
 
-    if (mApp.getState() == Application::APP_SYNCED_STATE)
-    {
-        mLoadTimer->expires_from_now(std::chrono::milliseconds(STEP_MSECS));
-        mLoadTimer->async_wait([this, cfg]() { this->generateLoad(cfg); },
-                               &VirtualTimer::onFailureNoop);
-    }
-    else
+    if (!mStartTime)
     {
-        CLOG_WARNING(
-            LoadGen,
-            "Application is not in sync, load generation inhibited. State {}",
-            mApp.getStateHuman());
-        mLoadTimer->expires_from_now(std::chrono::seconds(10));
-        mLoadTimer->async_wait(
-            [this, cfg]() { this->scheduleLoadGeneration(cfg); },
-            &VirtualTimer::onFailureNoop);
+        mStartTime =
+            std::make_unique<VirtualClock::time_point>(mApp.getClock().now());
     }
-}
 
-bool
-GeneratedLoadConfig::isDone() const
-{
-    return (isCreate() && nAccounts == 0) || (isLoad() && nTxs == 0) ||
-           (isSorobanSetup() && nInstances == 0);
-}
-
-bool
-GeneratedLoadConfig::areTxsRemaining() const
-{
-    return (isCreate() && nAccounts != 0) || (!isCreate() && nTxs != 0);
+    createRootAccount();
```

Comment:
Now that we have a single start function, we should change `if (!mRoot)` to `releaseAssert(!mRoot)` in `createRootAccount()`.